kernel/arch/dreamcast/include/dc/sq.h

```diff
@@ -53,20 +54,31 @@ __BEGIN_DECLS
 /** \brief   Set Store Queue QACR* registers
     \ingroup store_queues
 */
-#define SET_QACR_REGS(dest) \
+#define SET_QACR_REGS(dest0, dest1) \
     do { \
-        uint32_t val = ((uint32_t)(dest)) >> 24; \
-        QACR0 = val; \
-        QACR1 = val; \
+        QACR0 = ((uintptr_t)(dest0)) >> 24; \
+        QACR1 = ((uintptr_t)(dest1)) >> 24; \
     } while(0)
 
-/** \brief   Mask dest to Store Queue area
+/** \brief   Mask dest to Store Queue area as address
     \ingroup store_queues
 */
-#define SQ_MASK_DEST(dest) \
-        ((uint32_t *)(void *) \
-        (MEM_AREA_SQ_BASE |   \
-        (((uint32_t)(dest)) & 0x03ffffe0)))
+#define SQ_MASK_DEST_ADDR(dest) (MEM_AREA_SQ_BASE | ((uintptr_t)(dest) & 0x03ffffe0))
+
+/** \brief   Mask dest to Store Queue area as pointer
+    \ingroup store_queues
+*/
+#define SQ_MASK_DEST(dest) ((uint32_t *)(void *)SQ_MASK_DEST_ADDR(dest))
+
+/** \brief  Lock Store Queues
+    \ingroup store_queues
```

Comment:
```suggestion
    \ingroup store_queues
    
    Locks the store queues so that they cannot be used from another thread 
    until unlocked. 
    
    \warning
    This function is called automatically by the store queue API provided by KOS; 
    however, it must be called manually when driving the SQs directly from outside 
    of this API. 
    
    \sa sq_unlock()
```